crawler.py

```diff
@@ -1,32 +1,43 @@
+import requests
+
 from bs4 import BeautifulSoup
 
-import requests
 
-def parse():
+def crawl():
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 '
+                      '(KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36'
+    }
+
+    games = []
 
-	HEADERS = {
-		'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36'
-	}
+    page_number = 1
+    while True:
+        page_url = f'https://marketplace.xbox.com/en-US/Games/Xbox360Games?Page={page_number}'
 
-	max_page = 43
-	pages = []
+        print(f"Loading page #{page_number}: {page_url}")
+        response = requests.get(page_url,headers=headers)
+        print("Done")
 
-	for x in range(1, max_page + 1):
-		pages.append(requests.get('https://marketplace.xbox.com/en-US/Games/Xbox360Games?Page=' + str(x), headers = HEADERS))
```

Comment:
... also, your initial implementation has one pretty nasty flaw.

It parses _all_ pages, and only then prints anything. So in case you have to apply some little update — you have to wait like 2 min to see, if it works.

So I rewritten the code in a way to provide results instantly.